From 69b07bc8c6256d100f34aca9f34c990346581bfb Mon Sep 17 00:00:00 2001
From: Sebastien Guiriec <s-guiriec@ti.com>
Date: Wed, 2 Mar 2011 08:23:21 +0100
Subject: ASoC: OMAP ABE: Fix up FM Digital to use 32 bits mode for McBSP

Switch FM Digital to 32 bits mode for McBSP and ABE configuration.
Currently the driver is using 16 bits mode and there is a issue
between ABE and McBSP. Switching to 32 bits mode is solving the
problem. FM device should be configured in standard I2S master
mode.

Change-Id: Ifb54d383f9e9d044c6771b757aeba497cd7df353
Signed-off-by: Sebastien Guiriec <s-guiriec@ti.com>
---
 sound/soc/omap/omap-abe.c |    4 ++--
 sound/soc/omap/sdp4430.c  |   18 ++++++++++++++----
 2 files changed, 16 insertions(+), 6 deletions(-)

Index: kernel/sound/soc/omap/omap-abe.c
===================================================================
--- kernel.orig/sound/soc/omap/omap-abe.c	2011-04-01 18:30:53.000000000 +0900
+++ kernel/sound/soc/omap/omap-abe.c	2011-04-02 13:24:44.000000000 +0900
@@ -1118,14 +1118,14 @@
 				if (stream == SNDRV_PCM_STREAM_PLAYBACK) {
 					/* MM_EXT connection to McBSP 2 ports */
 					format.f = 48000;
-					format.samp_format = STEREO_RSHIFTED_16;
+					format.samp_format = STEREO_16_16;
 					abe_connect_serial_port(MM_EXT_OUT_PORT,
 								&format, MCBSP2_TX);
 					abe_dai_enable_data_transfer(MM_EXT_OUT_PORT);
 				} else {
 					/* MM_EXT connection to McBSP 2 ports */
 					format.f = 48000;
-					format.samp_format = STEREO_RSHIFTED_16;
+					format.samp_format = STEREO_16_16;
 					abe_connect_serial_port(MM_EXT_IN_PORT,
 								&format, MCBSP2_RX);
 					abe_dai_enable_data_transfer(MM_EXT_IN_PORT);
